server/src/main/java/io/unitycatalog/server/service/MetastoreService.java

```diff
@@ -0,0 +1,28 @@
+package io.unitycatalog.server.service;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.annotation.ExceptionHandler;
+import com.linecorp.armeria.server.annotation.Get;
+import io.unitycatalog.server.auth.UnityCatalogAuthorizer;
+import io.unitycatalog.server.auth.decorator.UnityAccessEvaluator;
+import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.persist.MetastoreRepository;
+import lombok.SneakyThrows;
+
+@ExceptionHandler(GlobalExceptionHandler.class)
+public class MetastoreService {
+  private static final MetastoreRepository METASTORE_REPOSITORY = MetastoreRepository.getInstance();
+
+  private final UnityAccessEvaluator evaluator;
+
+  @SneakyThrows
+  public MetastoreService(UnityCatalogAuthorizer authorizer) {
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
+
+  // TODO: should this endpoint be open to all?
```

Comment:
What auth do we have for other endpoints?

Reply:
These [privileges](https://github.com/unitycatalog/unitycatalog/blob/main/api/all.yaml#L2213-L2226) on these [securables](https://github.com/unitycatalog/unitycatalog/blob/main/api/all.yaml#L2202-L2211)

Reply:
Since Metastore is the top-level object, it might be okay to not have extra authorization checks for it. We will still have token authentication by default I presume for all APIs.

@tdas To confirm if this is okay.

Reply:
Not having auth. is okay. Remove the TODO